app/backend/core/ichatgptproxy.py

```diff
@@ -0,0 +1,32 @@
+from abc import abstractmethod
+
+class IChatGptProxy:
```

Comment:
Whats the reason for this abstract class? I see only one subclass of it currently.

Reply:
The idea was to allow developers to create their own proxy based on open ai models and different model versions. Also decouple approaches from model knowledge. Turns out it adds more complexity for other approaches. I have removed the changes and kept it simple as before.